packages/components/src/InlineLabel/InlineLabel.tsx

```diff
@@ -33,6 +35,13 @@ interface InlineLabelProps {
   children: ReactNode;
 }
 
+interface DismissableInlineLabelProps extends BaseInlineLabelProps {
+  dismissAriaLabel: string;
+  onDismiss(): void;
+}
+
+type InlineLabelProps = XOR<BaseInlineLabelProps, DismissableInlineLabelProps>;
+
```

Comment:
you can actually avoid the whole XOR thing by adding 1 prop with the `DismissableInlineLabelProps` as the type

For example

```tsx
interface InlineLabelProps {
  // label props here
  ...
  dismissAction?: DismissableInlineLabelProps;
}

interface DismissableInlineLabelProps {
  ariaLabel: string;
  onDismiss(): void;
}
```

and using it as 

```jsx
<InlineLabel dismissAction={{ ariaLabel: "", onDismiss: () => {} }} >
  Thing
</InlineLabel>
```

bonus point for not having to say `dismissAriaLabel`